### ISSUE TITLE
gradle build failiure

### ISSUE THREAD

**Issue by brobostigon** [nhsx/COVID-19-app-Android-BETA#2]
I just tried to build it, and got the following errors,

* Where:
Build file '/COVID-19-app-Android-BETA/app/build.gradle' line: 49
* What went wrong:
A problem occurred evaluating project ':app'.
> Missing property `sonar.baseUrl` please see README for instructions

After reading the readme it suggests a secret/property needs to be added, however doesn't specify what and or where that would be found to be added, and the same for the other 2 it mentions in the readme.

**Comment by mikemonteith** [comment]
You can add `sonar.baseUrl` and the other values mentioned in the README to the `gradle.properties` file. Just don't commit them!

**Comment by mikemonteith** [comment]
The readme also links here https://docs.gradle.org/current/userguide/build_environment.html for more info on how to set gradle properties

**Comment by brobostigon** [comment]
I read the readme and understood that as i mentioned, however not how the information to be entered into those is acquired and where from.

**Comment by mikemonteith** [comment]
Ah sorry I misunderstood. baseUrl is the URL of the backend server where the contact data is sent so it would be nice to have the backend open sourced too

**Comment by mikemonteith** [comment]
If you just want to get the build to work, you can use any nonsense values.

**Comment by brobostigon** [comment]
No worries. Ah, thank you for explaining.

**Comment by brobostigon** [comment]
What type of server does the backend need to be? Can it be something like my postgresql server?

**Comment by jimsmart** [comment]
> so it would be nice to have the backend open sourced too

+1 to this.

**Comment by robdyke** [comment]
> > so it would be nice to have the backend open sourced too
> 
> +1 to this.

This we have been promised....

**Comment by jorgecc** [comment]
If you want to miss to read the documentation, then you could edit the next line in build.gradle

```
def baseUrl = findProperty("sonar.baseUrl") ?: "https://www.google.com"
```

And you do the same with the other properties. It won't work but at least you could compile.

**Comment by halfninja** [comment]
The [gradle-properties-plugin](https://github.com/stevesaliman/gradle-properties-plugin)  is good for this - these properties could then be set in `gradle-local.properties`, which could be `.gitignore`d so it isn't committed.

**Comment by shantanukarve** [comment]
> You can add `sonar.baseUrl` and the other values mentioned in the README to the `gradle.properties` file. Just don't commit them!

The README says: "Please DO NOT add them to the project's gradle.properties." So I guess what it means is don't do a git add right  ? by the way, but adding them the configuration was successful.

**Comment by edent** [comment]
I'm pasting this message in every active GitHub issue, so you may receive duplicate notifications.

Today, I'm happy to announce that NHSX has released the full git commit history for the Isle of Wight Beta apps.

* Android - https://github.com/nhsx/COVID-19-app-Android-BETA
* iOS - https://github.com/nhsx/COVID-19-app-iOS-BETA

As discussed, we have redacted API keys, sensitive domain names, and some of the developers' personal details. I am still waiting on final approval to publish the server-side code.

I would like to personally thank the community for your comments, bug reports, and vulnerability disclosures. They all went into helping the development process.

The beta trial of this app has now ended and we've moved to the next phase of app development. It is our intention to publish the source code of future apps as the binaries are released to the public.

Once again, thank you for being part of this.

Terence Eden
Head of Open Technology - NHSX
